pkg/storage/queue.go

```diff
@@ -521,17 +521,28 @@ func (bq *baseQueue) processReplica(
 	defer cancel()
 	log.Eventf(ctx, "processing replica")
 
+	if destroyed := func() bool {
+		repl.mu.Lock()
+		defer repl.mu.Unlock()
+		return repl.mu.destroyed != nil
```

Comment:
If we put it in there, then it'll only run for the queues that require a replica lease, whereas my understanding is that it'd be beneficial to avoid processing destroyed replicas in all queues.

I've added a Replica.IsDestroyed method and switched to using that.